atc/exec/check_step.go

```diff
@@ -136,11 +136,9 @@ func (step *CheckStep) run(ctx context.Context, state RunState, delegate CheckDe
 	// Point scope to resource before check runs. Because a resource's check build
 	// summary is associated with scope, only after pointing to scope, check status
 	// can be fetched.
-	if step.plan.Resource != "" {
```

Comment:
Is this a bug fix for something else? Where we never update the `resource_config_id` of resource types because of this if statement?

Reply:
Yes, I think so. This is a bug introduced by #7208 (in-memory check) in 7.6.0. With this bug, `resource_config_id` is never updated in table `resource_types`.

Reply:
@clarafu BTW, #8049 is an optimization that avoids duplicate updating scopes.